src/Cli/dotnet/commands/dotnet-workload/install/WorkloadGarbageCollector.cs

```diff
@@ -131,35 +142,53 @@ void GarbageCollectWorkloadManifestsAndPacks()
+                {
+                    resolvers.Add((GetResolver(workloadSet), workloadSet, gcAction));
+                }
             }
 
-            foreach (var (resolver, workloadSet) in resolvers)
+            foreach (var (resolver, workloadSet, gcAction) in resolvers)
             {
                 foreach (var manifest in resolver.GetInstalledManifests())
                 {
                     _verboseReporter.WriteLine($"GC: Keeping manifest {manifest.Id} {manifest.Version}/{manifest.ManifestFeatureBand} as part of workload set {workloadSet}");
-                    ManifestsToKeep.Add((new ManifestId(manifest.Id), new ManifestVersion(manifest.Version), new SdkFeatureBand(manifest.ManifestFeatureBand)));
+
+                    var manifestKey = (new ManifestId(manifest.Id), new ManifestVersion(manifest.Version), new SdkFeatureBand(manifest.ManifestFeatureBand));
+                    GCAction existingAction;
+                    if (!_manifests.TryGetValue(manifestKey, out existingAction))
+                    {
+                        existingAction = GCAction.Collect;
+                    }
+
+                    //  Take the "greater" action, ie if a manifest would be KeepWithoutPacks with one resolver and Keep with another one, then it should be kept
+                    if (gcAction > existingAction)
+                    {
+                        _manifests[manifestKey] = gcAction;
+                    }
                 }
 
-                foreach (var pack in _installedWorkloads.SelectMany(workloadId => resolver.GetPacksInWorkload(workloadId))
-                    .Select(packId => resolver.TryGetPackInfo(packId))
-                    .Where(pack => pack != null))
+                if (gcAction == GCAction.Keep)
                 {
-                    _verboseReporter.WriteLine($"GC: Keeping workload pack {pack.ResolvedPackageId} {pack.Version} as part of workload set {workloadSet}");
-                    PacksToKeep.Add((new WorkloadPackId(pack.ResolvedPackageId), pack.Version));
+                    foreach (var pack in _installedWorkloads.SelectMany(workloadId => resolver.GetPacksInWorkload(workloadId))
+                        .Select(packId => resolver.TryGetPackInfo(packId))
+                        .Where(pack => pack != null))
+                    {
+                        _verboseReporter.WriteLine($"GC: Keeping workload pack {pack.ResolvedPackageId} {pack.Version} as part of workload set {workloadSet}");
+                        PacksToKeep.Add((new WorkloadPackId(pack.ResolvedPackageId), pack.Version));
+                    }
                 }
             }
 
-            //  NOTE: We should not collect baseline workload manifests. When we have a corresponding baseline manifest, this will happen, as we have logic
+            //  NOTE: We should not collect baseline workload manifests. When we have a corresponding baseline workload set, this will happen, as we have logic
```

Comment:
I think with your changes we now have working baseline workload sets.  I think until we switch to using workload sets by default we could still collect the baseline manifests, because the workload set won't be active unless you switch to workload set mode.